tasks/mrrc-collect-params/mrrc-collect-params.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: mrrc-collect-params
+spec:
+  params:
+    - name: dataJsonPath
+      description: path to data json file
+  results:
+    - name: mrrcParamFilePath
+      description: path of the env file for following tasks
+    - name: cosignPubKeyFile
+      description: the public key file name for cosign verification
+    - name: cosignPubKeySecret
+      description: the secret name for cosign verify public key
+    - name: charonAWSSecret
+      description: the secret name for charon aws credential file
+  steps:
+    - name: mrrc-collect-params
+      image: quay.io/konflux-ci/yq:latest
+      script: |
+        set -eux
+
+        cd $(workspaces.data.path)
+        mrrcEnvFile="./mrrc.env"
```

Comment:
I would suggest to use the same subdirectory as the data file is in. Similar to how we do it here: https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/create-pyxis-image/create-pyxis-image.yaml#L94

We store all of the data for the pipeline in a subdirectory. In the past this was needed because all of the pipeline runs were sharing a single PVC and so PLRs running in parallel would otherwise interfere.

We now have a new PVC for each PLR, so this is no longer a real issue, but it's better to be consistent. Eventually we will probably get rid of the subdirectories.

Reply:
Oh, ok. Will try to use this.

Reply:
Why use the snapshotSpec instead of the data file which is already a parameter to this task?

Reply:
@johnbieren I'm following @mmalina 's advice to use that "snapshotPath". I just copied an example usage which assign the value to this snapshotPath: https://github.com/konflux-ci/release-service-catalog/blob/development/pipelines/fbc-release/fbc-release.yaml#L138

Reply:
> Seems the "$(tasks.collect-data.results.snapshotSpec)" which is from the collect-data task passed an empty value there. Is something I'm missing for the usage of this?

Ok. After checking seems I missed to set the subdirectory param value for the collect-data. After fixing it can pass my local test.

Reply:
> Why use the snapshotSpec instead of the data file which is already a parameter to this task?

@johnbieren Oh ok I know what you mean. Will use the data path directly. Thanks.

Reply:
Fixed in new commit with a subdirectory setting with the dataJsonPath

Reply:
Yeah, what I meant is to use the same approach as in that other task, but here we can use the data path instead of the snapshot spec path.